pkg/asset/ignition/bootstrap/baremetal/template.go

```diff
@@ -111,14 +117,47 @@ func GetTemplateData(config *baremetal.Platform, networks []types.MachineNetwork
 	templateData.ExternalStaticDNS = config.BootstrapExternalStaticDNS
 	templateData.ExternalMACAddress = config.ExternalMACAddress
 
-	_, externalURLv6 := externalURLs(config.APIVIPs)
+	// If the user has manually set disableVirtualMediaTLS to False in the Provisioning CR,
+	// then enable TLS in ironic
+	protocol := "http"
+	provisioningCR := struct {
+		apiVersion string `yaml:"apiVersion"`
+		kind       string `yaml:"kind"`
+		metadata   struct {
+			name string `yaml:"name"`
+		} `yaml:"metadata"`
+		spec struct {
+			provisioningInterface     string `yaml:"provisioningInterface"`
+			provisioningIP            string `yaml:"provisioningIP"`
+			provisioningNetworkCIDR   string `yaml:"provisioningNetworkCIDR"`
+			provisioningNetwork       string `yaml:"provisioningNetwork"`
+			provisioningDHCPRange     string `yaml:"provisioningDHCPRange"`
+			provisioningOSDownloadURL string `yaml:"provisioningOSDownloadURL"`
+			watchAllNamespaces        bool   `yaml:"watchAllNamespaces"`
+			disableVirtualMediaTLS    bool   `yaml:"disableVirtualMediaTLS"`
+		} `yaml:"spec"`
+	}{}
+	provisioningCRFile, _ := os.Open("99_baremetal-provisioning-config.yaml") // which directory will this be in?
+	defer provisioningCRFile.Close()
+	if provisioningCRFile != nil {
+		decoder := yaml.NewDecoder(provisioningCRFile)
+		if err := decoder.Decode(&provisioningCR); err != nil {
+			logrus.Warnf("Error in decoding Provisioning CR YAML file: %s.", err)
+		}
+	}
+	if provisioningCR.spec.disableVirtualMediaTLS == false {
+		templateData.DisableIronicVirtualMediaTLS = false
```

Comment:
Any boolean value is false by default, you may want to set it to `true` somewhere else.

Reply:
Done